enf/dns_zone.go

```diff
@@ -0,0 +1,92 @@
+package enf
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// DNSService handles communication with the DNS related
+// methods of the ENF API. These methods are used to manage the DNS zones,
+// DNS records, and service endpoints on the ENF.
+type DNSService service
+
+// Zone represents a DNS zone within the ENF.
+type Zone struct {
+	Description    *string `json:"description"`
+	EnfDomain      *string `json:"enf_domain"`
+	ID             *string `json:"id"`
+	ZoneDomainName *string `json:"zone_domain_name"`
+}
+
+// CreateZoneRequest represents a request to create a DNS zone within the ENF.
+// Uses the same fields as the Zone struct.
+type CreateZoneRequest struct {
+	Description    *string `json:"description"`
+	ZoneDomainName *string `json:"zone_domain_name"`
+}
+
+// UpdateZoneRequest represents a request to update a DNS zone within the ENF.
+type UpdateZoneRequest struct {
+	Description *string `json:"description"`
+}
+
+type zoneResponse struct {
+	Data []*Zone                `json:"data"`
+	Page map[string]interface{} `json:"page"`
+}
+
+// ListDNSZones lists all the DNS zones for a given ENF domain (::/48 address).
+func (s *DNSService) ListDNSZones(ctx context.Context) ([]*Zone, *http.Response, error) {
```

Comment:
This is already on the DNS service, so would just `ListZones` be acceptable?  Ditto for all the other methods.